scripts/Diffraction/single_crystal/sxd.py

```diff
@@ -0,0 +1,320 @@
+        logger.notice(f"Indexed initially = {nindexed}/{npeaks} peaks within {tol_to_report}")
+        mantid.IndexPeaks(PeaksWorkspace=peaks_name, Tolerance=tol, CommonUBForAll=False, EnableLogging=False)
+
+        # optimize global shift in detector (equivalent to sample placement)
+        initialUB = np.copy(SXD.retrieve(peaks_name).sample().getOrientedLattice().getUB())
+        _, pos_table, *_ = mantid.OptimizeCrystalPlacement(
+            PeaksWorkspace=peaks_name,
+            ModifiedPeaksWorkspace="temp",
+            AdjustSampleOffsets=True,
+            OptimizeGoniometerTilt=True,
+            MaxIndexingError=tol,
+            MaxSamplePositionChangeMeters=maxShiftInMeters,
+            EnableLogging=False,
+        )
+        x, y, z = pos_table.cell(0, 1), pos_table.cell(1, 1), pos_table.cell(2, 1)
+        rotx, roty, rotz = pos_table.cell(3, 1), pos_table.cell(4, 1), pos_table.cell(5, 1)
+        for ibank in range(1, 12):
+            mantid.MoveInstrumentComponent(
+                Workspace=wsname, ComponentName="bank" + str(ibank), RelativePosition=True, X=-x, Y=-y, Z=-z, EnableLogging=False
+            )
+            mantid.ApplyInstrumentToPeaks(
+                InputWorkspace=peaks_name, InstrumentWorkspace=wsname, OutputWorkspace=peaks_name, EnableLogging=False
+            )
+        newUB = getR(rotx, [1, 0, 0]) @ getR(roty, [0, 1, 0]) @ getR(rotz, [0, 0, 1]) @ initialUB
+        mantid.SetUB(peaks_name, UB=newUB, EnableLogging=False)
+        # report progress
+        nindexed, *_ = mantid.IndexPeaks(PeaksWorkspace=peaks_name, Tolerance=tol_to_report, CommonUBForAll=False, EnableLogging=False)
+        logger.notice(f"Indexed after global instrument translation and rotation = {nindexed}/{npeaks} peaks within {tol_to_report}")
+        mantid.IndexPeaks(PeaksWorkspace=peaks_name, Tolerance=tol, CommonUBForAll=False, EnableLogging=False)
+
+        # optimize position of each bank independently
+        for ibank in range(1, 12):
+            logger.notice(f"Optimizing bank position {ibank}")
+            peaks_bank = peaks_name + "_bank"
+            mantid.FilterPeaks(
+                InputWorkspace=peaks_name,
+                OutputWorkspace=peaks_bank,
+                FilterVariable="RunNumber",
+                BankName="bank" + str(ibank),
+                EnableLogging=False,
+            )
+            # report inital indexing
+            npeaks_bank = SXD.retrieve(peaks_bank).getNumberPeaks()
+            nindexed, *_ = mantid.IndexPeaks(PeaksWorkspace=peaks_bank, Tolerance=tol_to_report, CommonUBForAll=False, EnableLogging=False)
+            logger.notice(f"Indexed before optimization = {nindexed}/{npeaks_bank} peaks within {tol_to_report}")
+            mantid.IndexPeaks(PeaksWorkspace=peaks_name, Tolerance=tol, CommonUBForAll=False, EnableLogging=False)
+            peaks_bank_mod, pos_table, *_ = mantid.OptimizeCrystalPlacement(
```

Comment:
After running the test SXD script I see a couple of workspaces left behind called CovarianceInfo or FitInfoTable which seem to be from OptimizeCrystalPlacement. If these are from the last bank processed they perhaps don't have any special significance. So could these be deleted?